packages/core/src/cloud/config.ts

```diff
@@ -0,0 +1,74 @@
+import type { PlatformApi } from "@llamaindex/cloud";
+import { BaseNode, TextNode } from "../Node.js";
+import { OpenAIEmbedding } from "../embeddings/OpenAIEmbedding.js";
+import type { TransformComponent } from "../ingestion/types.js";
+import { SimpleNodeParser } from "../nodeParsers/SimpleNodeParser.js";
+
+export type GetPipelineCreateParams = {
+  pipelineName: string;
+  pipelineType: PlatformApi.PipelineType;
+  transformations?: TransformComponent[];
+  inputNodes?: BaseNode[];
+};
+
+function getTransformationConfig(
+  transformation: TransformComponent,
+): PlatformApi.ConfiguredTransformationItem {
+  if (transformation instanceof SimpleNodeParser) {
+    return {
+      configurableTransformationType: "SIMPLE_FILE_NODE_PARSER",
+      component: {
+        includeMetadata: transformation.includeMetadata,
+        includePrevNextRel: transformation.includePrevNextRel,
+      },
+    };
+  }
+  if (transformation instanceof OpenAIEmbedding) {
+    return {
+      configurableTransformationType: "OPENAI_EMBEDDING",
+      component: {
+        modelName: transformation.model,
+        embedBatchSize: transformation.embedBatchSize,
+        dimensions: transformation.dimensions,
+      },
+    };
+  }
```

Comment:
@yisding @himself65 @EmanuelCampos, any alternative ideas for converting a transformation to its cloud config?

We could use a generic approach using `JSON.stringify` but then we might 
1. need to clean the serialized JSON afterwards AND
2. we might want to send parameters of sub-components to the cloud (e.g. chunksize of the textsplitter) which would be missing
- that's why I didn't do this

Adding a `toConfig` function to `TransformerComponent` is an alternative, but might be too much hassle.